recipes/sdbus-cpp/all/conanfile.py

```diff
@@ -0,0 +1,129 @@
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/Kistler-Group/sdbus-cpp"
+    description = "High-level C++ D-Bus library for Linux designed" \
+                  " to provide easy-to-use yet powerful API in modern C++"
+    topics = ("conan", "dbus", "sd-bus", "sdbus-c++", "sdbus-cpp")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_code_gen": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_code_gen": False,
+    }
+    exports_sources = "CMakeLists.txt"
+    generators = ("cmake", "pkg_config")
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    @property
+    def _minimum_cpp_standard(self):
+        return 17
+
+    @property
+    def _minimum_compilers_version(self):
+        return {
+            "gcc": "7",
+            "clang": "6",
+        }
+
+    def configure(self):
+        if self.settings.os != "Linux":
+            raise ConanInvalidConfiguration("Only Linux supported")
+        if self.options.shared:
+            del self.options.fPIC
+
+        if self.settings.compiler.get_safe("cppstd"):
+            tools.check_min_cppstd(self, self._minimum_cpp_standard)
+        min_version = self._minimum_compilers_version.get(str(self.settings.compiler))
+        if not min_version:
+            self.output.warn("{} recipe lacks information about the {} compiler support.".format(
+                self.name, self.settings.compiler))
+        else:
+            if tools.Version(self.settings.compiler.version) < min_version:
+                raise ConanInvalidConfiguration("{} requires C++{} support. The current compiler {} {} does not support it.".format(
+                    self.name, self._minimum_cpp_standard, self.settings.compiler, self.settings.compiler.version))
+
+    def build_requirements(self):
+        self.build_requires("pkgconf/1.7.3")
+        if self.options.with_code_gen:
+            self.build_requires("expat/2.2.10")
+
+    def requirements(self):
+        self.requires("libsystemd/247.2")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        src_folder = "{}-{}".format(self.name, self.version)
+        os.rename(src_folder, self._source_subfolder)
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.definitions["BUILD_CODE_GEN"] = self.options.with_code_gen
+        self._cmake.definitions["BUILD_DOC"] = False
+        self._cmake.definitions["BUILD_TESTS"] = False
+        self._cmake.definitions["BUILD_LIBSYSTEMD"] = False
+        self._cmake.configure()
+        return self._cmake
+
+    def fix_tools_cmake(self):
```

Comment:
```suggestion
    def _patch_sources(self):
```

Any custom method should be declared as protected to avoid future name collisions.

_patch_sources is no official name or a pattern, but it's  the most popular in CCI.